tests/android/androidTest.ts

```diff
@@ -65,4 +65,24 @@ export const androidTest = baseTest.extend<PageTestFixtures, AndroidWorkerFixtur
     const page = await androidContext.newPage();
     await run(page);
   },
+
+  rewriteRequestToLocalhost: [async ({ isAndroid, rewriteRequestURLToLocalhost }, use) => {
+    await use(request => {
+      if (!isAndroid)
+        return request;
+      const initializer = (request as any)._initializer;
+      initializer.url = rewriteRequestURLToLocalhost(initializer.url);
```

Comment:
Let's disable all tests that need this. I don't think that tampering with internals is worth it.